homeworks/07-js-basics-2/script.js

```diff
@@ -0,0 +1,215 @@
+  for (let i = 0; i < temp.length; i++) {
+    if (temp[i].length > 1 && Math.abs(temp[i][0] - temp[i][1]) == 1) {
+      counter++;
+    }
+  }
+  return counter;
+}
+
+//  task 3 https://www.codewars.com/kata/5aba780a6a176b029800041c
+
+function maxMultiple(divisor, bound) {
+  for (let i = bound; i > 0; i--) {
+    if (i % divisor == 0) {
+      return i;
+    }
+  }
+}
+
+//  task 4 https://www.codewars.com/kata/514a6336889283a3d2000001
+
+function getEvenNumbers(numbersArray) {
+  return numbersArray.filter((e) => e % 2 == 0);
+}
+
+//  task 5 https://www.codewars.com/kata/5a090c4e697598d0b9000004
+
+function solve(arr) {
+  let numbers = [];
+  arr.sort((a, b) => b - a);
+  let length = arr.length;
+  for (let i = 0; i < length / 2; i++) {
+    let firstEl = arr.shift();
+    numbers.push(firstEl);
+    firstEl = 0;
+    let lastEl = arr.pop();
+    if (lastEl) {
+      numbers.push(lastEl);
+    }
+  }
+  return numbers;
+}
+
+//  task 6 https://www.codewars.com/kata/566044325f8fddc1c000002c
+
+function evenChars(string) {
+  let arr = Array.from(string);
+  let length = arr.length;
+  if (100 > length && length > 2) {
+    return arr.filter((_, i) => {
+      return i % 2 == 1;
+    });
+  }
+  return "invalid string";
+}
+
+//  task 7 https://www.codewars.com/kata/545a4c5a61aa4c6916000755
+
+function gimme(triplet) {
+  let min = Math.min(...triplet);
+  let max = Math.max(...triplet);
+  let midNumber = (element) => min < element && element < max;
+  return triplet.findIndex(midNumber);
+}
+
+//  task 8 https://www.codewars.com/kata/578553c3a1b8d5c40300037c
+
+const binaryArrayToNumber = (arr) => {
+  let str = arr.join("");
+  return parseInt(str, 2);
+};
+9;
+
+//  task 9 https://www.codewars.com/kata/585d7d5adb20cf33cb000235
+
+function findUniq(arr) {
+  let str = arr[0];
+  let x = arr.filter((e) => {
+    return e != str;
+  });
+  if (x.length === 1) {
+    return x[0];
+  }
+  return str;
+}
+
+//  task 10 https://www.codewars.com/kata/581e014b55f2c52bb00000f8
+
+function decipherThis(str) {
+  let arr = str.split(" ");
+  for (let i = 0; i < arr.length; i++) {
+    let tempArr = arr[i];
+    let number = tempArr.match(/\d+/g);
+    let str = "";
+    str = String.fromCharCode(number[0]);
+    if (tempArr.length !== 1) {
+      tempArr = tempArr.replace(tempArr.match(/\d+/g)[0], "");
+      let second = tempArr.length > 0 ? tempArr[0] : "";
+      let last = tempArr.length > 0 ? tempArr[tempArr.length - 1] : "";
+      str += last;
+      if (tempArr.length >= 2) {
+        str += tempArr.substring(1, tempArr.length - 1);
+      }
+      if (tempArr.length != 1 && second) {
+        str += second;
+      }
+    }
+    arr[i] = [];
+    arr[i] = str;
+  }
+  str = arr.join(" ");
+  return str;
+}
+
+//  task 11 https://www.codewars.com/kata/578aa45ee9fd15ff4600090d
+
+function sortArray(array) {
+  let arr = array.filter((e, i) => e % 2 != 0).sort((a, b) => a - b);
```

Comment:
You have mutated incoming array!
Create a copy and modify and return it.
Incoming data should stay safe.